reflect.go

```diff
@@ -208,28 +225,32 @@ func (r *Reflector) Reflect(v interface{}) *Schema {
 
 // ReflectFromType generates root schema
 func (r *Reflector) ReflectFromType(t reflect.Type) *Schema {
+	if t.Kind() == reflect.Ptr {
+		t = t.Elem() // re-assign from pointer
+	}
+
+	// Set default base schema ID
+	if r.BaseSchemaID == "" {
+		r.BaseSchemaID = ID("https://" + t.PkgPath())
```

Comment:
PkgPath() can return an empty string in some cases. Is it worth checking for that here, and either returning an error, or perhaps panicking, as it probably means this function is used improperly

Reply:
Good point. This shouldn't happen as `PkgPath()` normally only returns empty on pointers and system objects. It would be good to have a guard case though to either not set the `BaseSchemaID`, or raise an error.

Reply:
That should be fixed now. There is proper ID validation and of course, this code was replacing the configured base schema ID, which was not good!